lint/linter/test-spec-urls.ts

```diff
@@ -49,19 +54,68 @@ const specsExceptions = [
   'https://github.com/WebAssembly/js-promise-integration',
 ];
 
-const allowedSpecURLs = [
-  ...(specData
-    .filter((spec) => spec.standing == 'good')
-    .map((spec) => [
-      spec.url,
-      spec.nightly?.url,
-      ...(spec.nightly ? spec.nightly.alternateUrls : []),
-      spec.series.nightlyUrl,
-    ])
-    .flat()
-    .filter((url) => !!url) as string[]),
-  ...specsExceptions,
-];
+interface ValidSpecHosts {
+  url: string;
+  alternateUrl: string;
+}
+
+const validSpecHosts: ValidSpecHosts[] = [];
+
+/**
+ * Get valid specification URLs from webref ids
+ * @returns array of valid spec urls (including fragment id)
+ */
+const getValidSpecURLs = async (): Promise<string[]> => {
+  const indexFile = await fetch(
+    'https://raw.githubusercontent.com/w3c/webref/main/ed/index.json',
+  );
+  const index = JSON.parse(await indexFile.text());
+  const webrefFiles: string[] = [];
+
+  index.results.forEach((spec) => {
+    if (spec.standing === 'good') {
+      if (spec.shortname === 'webnn') {
+        validSpecHosts.push({
+          url: spec.series.releaseUrl,
+          alternateUrl: spec.nightly?.url,
+        });
+      } else {
+        validSpecHosts.push({
+          url: spec.series.nightlyUrl,
+          alternateUrl: spec.nightly?.url,
+        });
+      }
+      if (spec.dfns) {
+        webrefFiles.push(spec.dfns);
+      }
+      if (spec.headings) {
+        webrefFiles.push(spec.headings);
+      }
+    }
+  });
+
+  const specURLsWithFragments: string[] = [];
+
+  const responses = await Promise.all(
+    webrefFiles.map(async (file) => {
+      const res = await fetch(
+        `https://raw.githubusercontent.com/w3c/webref/main/ed/${file}`,
+      );
+      const type = file.match(/^([^/]+)\//)?.[1] ?? 'dfns';
+      const json = await res.json();
+      const dfns = json[type];
+      return dfns.map((entry) => {
+        const url = new URL(entry.href);
+        return url.origin + url.pathname + decodeURIComponent(url.hash);
+      });
```

Comment:
Dfns extracts now also contain anchors that appear in sections that are "For web developers". The anchors complete the underlying definition under a `links` property. The code needs to be adjusted accordingly to also take them into account. Something like:

```suggestion
      return dfns
        .map((entry) => [entry].concat(entry.links ?? []))
        .flatMap(links => links.map(link => {
          const url = new URL(link.href ?? link);
          return url.origin + url.pathname + decodeURIComponent(url.hash);
        }));
```